### ISSUE TITLE
WeeChat scripts not found

### ISSUE THREAD

**Issue by jluttine** [NixOS/nixpkgs#89499]
**Describe the bug**

WeeChat Python scripts aren't found in WeeChat. (It is very much possible that I'm doing something wrong. I tried to follow these instructions to the best of my understanding: https://nixos.org/nixpkgs/manual/#sec-weechat )

**To Reproduce**
1. Put this in `configuration.nix`:
```
environment.systemPackages = with pkgs; [
  weechat
];
nixpkgs.overlays = [
  (
    self: super: {
      weechat = super.weechat.override {
        configure = {availablePlugins, ...}: {
          scripts = with pkgs.weechatScripts; [
            weechat-matrix
            wee-slack
          ];
          plugins = builtins.attrValues availablePlugins;
        };
      };
    }
  )
];
```
2. Rebuild `sudo nixos-rebuild test`
3. Run `weechat`
4. In WeeChat, try to load weechat-matrix plugin: `/script load matrix.py`. WeeChat gives the following error: `python: script "matrix.py" not found`

**Expected behavior**

I would have expected weechat-matrix to load and then all the following steps to work: https://github.com/poljar/weechat-matrix#configuration

**Notify maintainers**

cc: @tilpner @lheckemann et al

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.41, NixOS, 20.09.git.0f5ce2fac0c (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - nixpkgs: `/etc/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: weechat weechat-matrix
# a list of nixos modules affected by the problem
module:
```

**Comment by Ma27** [comment]
Just did a brief check in my nixpkgs checkout with an expression like this:

``` nix
with import ./. { };
weechat.override {
  configure = {availablePlugins, ...}: {
    scripts = with pkgs.weechatScripts; [
      weechat-matrix
      wee-slack
    ];
    plugins = builtins.attrValues availablePlugins;
  };
}
```

The `wee-slack` module seems to load fine, but the `weechat-matrix` module isn't and WeeChat reports the following error:

```
         │00:10:52     | python: loading script "/nix/store/bq3nwr9bg52lilbja6s4h2nkhzwj0n97-python3.7-weechat-matrix-0.1.0/share/matrix.py"
         │00:10:52     | python: stdout/stderr (?): Traceback (most recent call last):
         │00:10:52     | python: stdout/stderr (?):   File "/nix/store/bq3nwr9bg52lilbja6s4h2nkhzwj0n97-python3.7-weechat-matrix-0.1.0/share/matrix.py", line 41, in <module>
         │00:10:52     | python: stdout/stderr (?):     import logbook
         │00:10:52     | python: stdout/stderr (?): ModuleNotFoundError: No module named 'logbook'
         │00:10:52 =!= | python: unable to parse file "/nix/store/bq3nwr9bg52lilbja6s4h2nkhzwj0n97-python3.7-weechat-matrix-0.1.0/share/matrix.py"
```

I assume that the `weechat-matrix` bridge contains e.g. an incomplete wrapper. ccing maintainers @tilpner @emilazy

**Comment by tilpner** [comment]
Some other scripts (including wee-slack) patch themselves to insert their dependencies into the script environment:

https://github.com/NixOS/nixpkgs/blob/fb6c3cee64826494973fdf4eb362eecac3fd8a11/pkgs/applications/networking/irc/weechat/scripts/wee-slack/default.nix#L15-L21

https://github.com/NixOS/nixpkgs/blob/fb6c3cee64826494973fdf4eb362eecac3fd8a11/pkgs/applications/networking/irc/weechat/scripts/wee-slack/libpath.patch#L8-L10

weechat-matrix currently does not, and thus the propagatedBuildInputs (which include Logbook) are not actually propagated.

Pending a proper fix (ideally without patching), you can do this:

```nix
{ weechat, weechat-matrix }: 
weechat.override {
  configure = {availablePlugins, ...}: {
    # this part instructs weechat to load the script, but does not bring its dependencies
    scripts = [ weechat-matrix ];
    plugins = [
      # the dependencies are propagated here
      (availablePlugins.python.withPackages (_: [ weechat-matrix ]))
    ];
  };
}
```

**Comment by alyssais** [comment]
PR to fix weechat-matrix: #110866.
